### ISSUE TITLE
Double-width tilde and wave dash in hashtag

### ISSUE THREAD

**Issue by igrep** [tootsuite/mastodon#15323]
### Expected behaviour

Both `#トロピカル～ジュ` and `#トロピカル〜ジュ` should be recognized as a single tag.

### Actual behaviour

Only `トロピカル` is recognized as a hash tag. Characters after "～" (double-width tilde) or "〜" (wave dash) are not recognized as the part of the tag.

i.e.
![image](https://user-images.githubusercontent.com/227057/102000906-8cce0000-3d2f-11eb-9efa-4fdb38647f6c.png)

(The original toot of the screenshot above is https://precure.ml/web/statuses/105370321446630212)

### Steps to reproduce the problem

Toot with a hash tag containing "～" (double-width tilde) or "〜" (wave dash).

- E.g. `#トロピカル～ジュ` or `#トロピカル〜ジュ`

### Specifications

- Tested server: v3.2.1
- Tested browser: Firefox ver. 83.0 on Windows 10 20H2.

### Background

Both "～" (double-width tilde) and "〜" (wave dash) are sometimes used to indicate "chouon" （長音）(a long or drawn-out vowel character) in Japanese and Chinese. This is similar to "ー" (Katakana-Hiragana Prolonged Sound Mark), but "～" (double-width tilde) and "〜" (wave dash) has more relaxed nuance.

For example, when we Japanese (and perhaps Chinese too) see `あー` (pronounced as "ah". In Roma-ji: ā) in a speech of some comic or novel, we imagine he/she is nodding, or just shouting if an exclamation mark followes (i.e. `あー！`). On the other hand, `あ～` or `あ～！` makes us imagine that he/she is relaxed ☺️ , such as he/she can't help uttering by taking a hot spring ♨️ , by having a massage, 💆  or by having a delicious foods 🍜 , etc.

Anyway, in other words, "～" (double-width tilde) or "〜" (wave dash) is a relaxed variation of "chouon", "ー" (Katakana-Hiragana Prolonged Sound Mark). So **both of them should be counted as a character consisting of a word**.

#### Reference

- About the general uses of "〜" (wave dash) https://en.wikipedia.org/wiki/Japanese_punctuation#Wave_dash
    - This refers only to "〜" (wave dash). But we also actually use "～" (double-width tilde) for those uses. See the next page below for details.
- About the confusing difference between "～" (double-width tilde) and "〜" (wave dash): https://en.wikipedia.org/wiki/Tilde#Unicode_and_Shift_JIS_encoding_of_wave_dash
- About "ー" （長音符）: https://en.wikipedia.org/wiki/Ch%C5%8Donpu
